iam/src/providers/coinPossession.ts

```diff
@@ -0,0 +1,119 @@
+// ----- Types
+import type { Provider, ProviderOptions } from "../types";
+import type { RequestPayload, VerifiedPayload } from "@gitcoin/passport-types";
+
+// ----- Ethers library
+import { Contract } from "ethers";
+import { formatUnits } from "@ethersproject/units";
+import { StaticJsonRpcProvider } from "@ethersproject/providers";
+
+/*
+Coin Possession Provider can be used to check a greater than balance for ethereum or any other evm token (ERC20).
+By default this will verify the ethereum balance for the address in the parameter. To customize the
+token set the contract_address in the options passed to the class.
+*/
+
+// ERC20 ABI
+const ERC20_ABI = [
+  {
+    constant: true,
+    inputs: [
+      {
+        name: "_owner",
+        type: "address",
+      },
+    ],
+    name: "balanceOf",
+    outputs: [
+      {
+        name: "balance",
+        type: "uint256",
+      },
+    ],
+    payable: false,
+    stateMutability: "view",
+    type: "function",
+  },
+];
+
+// set the network rpc url based on env
+export const RPC_URL = process.env.RPC_URL;
+
+export async function getTokenBalance(address: string, tokenContractAddress: string): Promise<number> {
+  // define a provider using the rpc url
+  const provider: StaticJsonRpcProvider = new StaticJsonRpcProvider(RPC_URL);
+  // load Token contract
+  const readContract = new Contract(tokenContractAddress, ERC20_ABI, provider);
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
+  const gtcBalance: string = await readContract?.balanceOf(address);
+  const balanceFormatted: string = formatUnits(gtcBalance, 18);
```

Comment:
I would rename this var, to be more generic as it refers to any token balance:

```suggestion
  const tokenBalance: string = await readContract?.balanceOf(address);
  const balanceFormatted: string = formatUnits(tokenBalance, 18);
```